astropy/cosmology/parameter.py

```diff
@@ -210,8 +212,104 @@ def register(fvalidate):
 
     # -------------------------------------------
 
+    def _get_init_arguments(self, repr=True):
+        """Initialization arguments.
+
+        Parameters
+        ----------
+        repr : bool
+            Whether to process the output into a more human-readable form that
+            also is more likely to round-trip with ``eval(repr(...))``.
+            Default `True`.
+
+        Returns
+        -------
+        dict[str, Any]
+        """
+        # The keys are added in this order because `repr` prints them in order.
+        kw = {"derived": self.derived,
+              "unit": self.unit,
+              "equivalencies": self.equivalencies}
+        # Validator is always turned into a function, but for ``repr`` it's nice
+        # to know if it was originally a string.
+        if repr and self._validate_repr in self._registry_validators:
```

Comment:
Really, this is all fine, but on second thought I would in `__init__` do `self._fvalidate_in = fvalidate` (before interpretation, so one can get rid of the class attribute) and then here you can just do as part of the `dict` definition,
```
"fvalidate": self._fvalidate_in if repr else self.fvalidate
```

or, with the suggested rename,
```
"fvalidate" : self.fvalidate if interpreted else self._fvalidate_in
```

Reply:
I do think that's slightly cleaner. I've made the suggested change, only calling the argument ``processed``, not ``interpreted``. Thanks!